### PR TITLE
Adding S3 support.

### PR DESCRIPTION
**Pull request by jlm3575**

### CODE REVIEW

File: tasks/tarball.yml [hunk 1]
```diff
@@ -5,9 +5,31 @@
 - name: "Ensure the zookeeper dir exists at {{zookeeper_dir}}"
   file: path={{zookeeper_dir}} state=directory
 
+# If the file exists already, the S3 call can experience errors
+- name: check whether the tarball is downloaded already
+  stat: path={{zookeeper_tarball_dir}}/zookeeper-{{zookeeper_version}}.tar.gz
+  register: zookeeper_tarball_script
+
+- name: Download zookeeper from S3
```
Comment (jlm3575) [inline tasks/tarball.yml]: Provide the option of storing the tarball in Amazon S3.

File: tasks/tarball.yml [hunk 2]
```diff
@@ -5,9 +5,31 @@
...
+    mode: get
+    bucket: "{{zookeeper_archive_s3_bucket}}"
+    object: "{{zookeeper_archive_s3_object}}"
+    dest: "{{zookeeper_tarball_dir}}/zookeeper-{{zookeeper_version}}.tar.gz"
+  register: zookeeper_s3_tarball_downloaded
+  tags: bootstrap
+  when: >
+    zookeeper_archive_s3_bucket is defined and
+    zookeeper_archive_s3_object is defined and
+    not zookeeper_tarball_script.stat.exists
```
Comment (jlm3575) [inline tasks/tarball.yml]: We have to make sure all vars are defined, and that the tarball hasn't already been downloaded.  The S3 directive will fail in some cases if the file exists.

File: tasks/tarball.yml [hunk 3]
```diff
@@ -5,9 +5,31 @@
...
+
 - name: Download zookeeper version.
   get_url: url={{zookeeper_url}} dest={{zookeeper_tarball_dir}}/zookeeper-{{zookeeper_version}}.tar.gz
   tags: bootstrap
+  register: zookeeper_tarball_downloaded
+  when: >
+    zookeeper_archive_s3_bucket is not defined or
```
Comment (jlm3575) [inline tasks/tarball.yml]: If they missed either var, download from Apache.

**Comment by ernestas-poskus** [comment]
Hi please rebase on master, build is now fixed with https://github.com/AnsibleShipyard/ansible-zookeeper/pull/67

**Comment by jlm3575** [comment]
@ernestas-poskus, rebased.

**Comment by ernestas-poskus** [comment]
LGTM, thank you contribution

**Pull request merged**